File: GptFactChecker/GptFactCheckerApi/Model/ClaimCheck.cs
namespace GptFactCheckerApi.Model;

public class ClaimCheck
{
    public string Id { get; set; }
    public string UserId { get; set; }
    public string Label { get; set; }
    public string ClaimCheckText { get; set; }
    public string DateCreated { get; set; }
    public List<string> References { get; set; } = new();
}

File: GptFactChecker/gpt-fact-checker-front/src/model/ClaimCheck.ts
import type ClaimCheckReaction from "./ClaimCheckReaction";

class ClaimCheck {
    public id: string;
    public userId: string;
    public label: string;
    public claimCheckText: string;
    public dateCreated: string;
    public references?: string[];

    public claimCheckReactions?: ClaimCheckReaction[];

    constructor() {
        this.id = '';
        this.userId = '';
        this.label = '';
        this.claimCheckText = '';
        this.dateCreated = new Date().toISOString(); //YYYY-MM-DDTHH:mm:ss.sssZ
    }
}

export default ClaimCheck;